src/gordon_gcp/plugins/janitor/reconciler.py

```diff
@@ -264,13 +290,13 @@ def _parse_rrset_message(self, message):
         missing_rrsets = [
             rs for rs in desired_rrsets if rs not in actual_rrsets
         ]
+        # don't try to add root SOA/NS records
+        missing_rrsets = self._remove_soa_and_root_ns(zone, missing_rrsets)
```

Comment:
I missed this PR, but maybe this should happen against `actual_rrsets` instead, since these are actual rrsets that we don't care about it. Obviously if someone wants to create NS or SOA (i.e. in `desired_rrsets`) that won't work, but maybe we should be hard failing in that case, instead of silently accepting but ignoring them.